books-refractored.js

```diff
@@ -0,0 +1,675 @@
+  });
+});
+
+//
+//----Year Style
+yearBtns.forEach((yearBtn) => {
+  yearBtn.addEventListener("mouseover", () => {
+    if (!yearBtn.classList.contains("selected")) {
+      yearBtn.classList.add("hover");
+    }
+  });
+
+  yearBtn.addEventListener("mouseout", () => {
+    yearBtn.classList.remove("hover");
+  });
+
+  yearBtn.addEventListener("click", () => {
+    yearBtn.classList.remove("hover");
+    toggleYearSelected(yearBtn);
+  });
+});
+//
+//----Year Filter event listense
+
+yearBtns.forEach((yearBtn) => {
+  yearBtn.addEventListener("click", (event) => {
+    switch (event.target.id) {
+      case "early-nineteen":
+        startYear = 1800;
+        endYear = 1849;
+        getYear(1800, 1849);
+        break;
+
+      case "late-nineteen":
+        startYear = 1850;
+        endYear = 1899;
+        getYear(1850, 1899);
+        break;
+
+      case "early-twenty":
+        startYear = 1900;
+        endYear = 1949;
+        getYear(1900, 1949);
+        break;
+
+      case "late-twenty":
+        startYear = 1950;
+        endYear = 1999;
+        getYear(1950, 1999);
+        break;
+
+      case "present":
+        startYear = 2000;
+        endYear = 2049;
+        getYear(2000, 2049);
+        break;
+    }
+    filteredBooks = yearFilteredBooks;
+    filterType = "Year";
+    isYearFiltered = true;
+    checkFiltered();
+    console.log(event.target.innerHTML);
+    applySorting();
+  });
+});
+
+//
+//----Sort Style
+
+sortBtns.forEach((sortBtn) => {
+  sortBtn.addEventListener("mouseover", () => {
+    if (!sortBtn.classList.contains("selected")) {
+      sortBtn.classList.add("hover");
+    }
+  });
+
+  sortBtn.addEventListener("mouseout", () => {
+    sortBtn.classList.remove("hover");
+  });
+
+  sortBtn.addEventListener("click", () => {
+    console.log(isSorted);
+    if (!isSorted) {
+      sortBtn.classList.remove("hover");
+      toggleSortSelected(sortBtn);
+    } else {
+      toggleSortSelected(sortBtn);
+    }
+  });
+});
+
+//
+//----Sort evenet listener
+
+highestRating.addEventListener("click", () => {
+  checkFiltered();
+  checkSorted(sortByHighRating);
+});
+
+newestPublished.addEventListener("click", () => {
+  checkFiltered();
+  checkSorted(sortByPublish);
+});
+
+author.addEventListener("click", () => {
+  checkFiltered();
+  checkSorted(sortByAuthor);
+});
+
+titleAZ.addEventListener("click", () => {
+  checkFiltered();
+  checkSorted(sortByTileAZ);
+});
+
+titleZA.addEventListener("click", () => {
+  checkFiltered();
+  checkSorted(sortByTileZA);
+});
+
+//---- General Style
+randomBookBtn.addEventListener("click", () => {
+  getRandomBook();
+  randomBookBtn.classList.add("selected");
+  setTimeout(() => {
+    randomBookBtn.classList.remove("selected");
+  }, 1000);
+  resetFunc();
+});
+
+reset.addEventListener("click", () => {
+  reset.classList.add("selected");
+  setTimeout(() => {
+    reset.classList.remove("selected");
+  }, 800);
+  bookCards(books);
+  resetFunc();
+});
```

Comment:
Very clever to add a reset button to clear the user input and also adding a short delay for visual clarity. 👏